src/peft/peft_model.py

```diff
@@ -234,8 +258,9 @@ def from_pretrained(
 
     def _setup_prompt_encoder(self, adapter_name: str):
         config = self.peft_config[adapter_name]
-        self.prompt_encoder = torch.nn.ModuleDict({})
-        self.prompt_tokens = {}
+        if not hasattr(self, "prompt_encoder"):
```

Comment:
This fixes a bug with prompt encoder models. Whenever a user calls `model.add_adapter(xxx)` currently in the main branch it will create a new module dict, thus erasing the previous one.

Reply:
So is this code a fix to the bug you mentioned, or a workaround that can be removed once the bug is fixed? If the latter, please add a `# TODO` with a comment that this can be removed when the bug is fixed.

Reply:
This should be the fix for the bug I have mentioned !

Reply:
Perfect